### DISCUSSION TITLE
Option to override dim and set it to 100% if beatmap to be played has no video

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#25003] [Ideas]
I propose an option in the settings below "Background dim" (and maybe also for the Visual Settings on the right, right before a beatmap starts) called something like "Set dim to 100% if beatmap has no video" which, if enabled, temporarily overrides the background dim option and makes it 100% if the beatmap played has no video. The actual value of the "Background dim" option stays the same.

This is useful because when the beatmap has no video anyway then I don't really need the thumbnail in the background while I'm playing because it really is only distracting more than anything and I can see the thumbnail in the beatmap selection and before the beatmap starts so I already consumed all "visual content" the beatmap has to offer so I don't really need to see it during gameplay.
Of course not everyone feels that way and that's why it should be an option rather than the default. And it should probably be off by default.

Usually I have dim set to 100% anyway but sometimes I just feel like playing my beatmaps and see which ones have video and watch that in the background for a more immersive experience but for the beatmaps that don't have video I don't really need the thumbnail in the background as it's just distracting and I'd prefer 100% in that case.

**Comment by Rekunan** [comment]
I feel like this would be better implemented as "if `video`, do `x`; else, do `y`" but that also sounds like opening a whole can of worms itself. Maybe as some sort of config file?

**Comment by peppy** [comment]
Seems too specific to the user's case. Very unlikely we will implement something like this. The closest would probably be storing per beatmap the dim state, as per stable (still requires manual intervention per beatmap by the user).
